README.md

```diff
@@ -115,6 +115,39 @@ Server
 In many rails projects when you run the server with `rails s` Rails sets RAILS_ENV to "development".  If you do that with this app, you will connect to the development database on AWS (if you have permission to decrypt the value).  Instead, run `RAILS_ENV=local rails s` to start the server and `RAILS_ENV=local rails c` to run the console.
 
 
+Multithreading
+========================
+Our multithreading functionality is done through ActiveJob, which employs DelayedJob on the backend.
+ActiveJob gets better after rails 5.2, so keep in mind that there is some functionality that is not perfect, until we can upgrade rails.
+
+Backstory:  Rails runs on a single thread.  Even when you schedule asynchronous operations, s.a. sending a request with HTTParty, that request is executed on the same thread as the rest of the application.  Rails simulates multithreading with creative use of scheduling.
+
+To put something on its own thread, you have to first create the second thread by starting the second worker on the same EC2 server:
+```
+RAILS_ENV=qa bin/delayed_job start
+```
+
+There are several options for implementing the multithreading backend in the code.  Sidekiq and DelayedJob are two of the more popular ones.  We chose DelayedJob, for its relative simplicity.  Each backend implementation can be used on its own, or through ActiveJob hooks (https://edgeguides.rubyonrails.org/active_job_basics.html).
+
+It's a good idea to use ActiveJob.  Even though using DelayedJob directly can give more powerful functionality, using the ActiveJob intermediary will allow future maintainers to switch implementations without rewriting app code.
+
+How do you call a method asynchronously?  Call it like we do here:
+```
+FindAvailableUserBarcodeJob.perform_later(user: self)
+```
+Perform, and perform_later calls are scheduled to go at some schedule you've set up.  
+At this point, if you have not started your second worker (remember delayed_job start?), your call will go on your regular app stack.  Multithreading will not happen.  If you have a second worker going, your asynchronous call will be put on that second worker's thread.  If you have multiple asynchronous calls going at the same time, they will be put on the second worker thread, but be single-threaded (schedule one after another) within that thread's stack.
```

Comment:
So for example, here, instead of "Multithreading will not happen" I'd say "the method will be processed synchronously".

Reply:
same as above